src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -0,0 +1,210 @@
+                        }
+
+                        var (children, count) = ConvertTo(filePath, headingBlocks, i + 1);
+                        item.Items = children;
+                        i += count;
+                        childrenCount += count;
+                    }
+
+                    if (i + 1 < headingBlocks.Length && headingBlocks[i + 1].Level < currentLevel)
+                    {
+                        break;
+                    }
+                }
+            }
+            while (++i < headingBlocks.Length);
+
+            return (items, items.Count + childrenCount);
+
+            TableOfContentsInputItem GetItem(HeadingBlock block)
+            {
+                if (block.Inline == null || !block.Inline.Any())
+                {
+                    return null;
+                }
+
+                string name = null;
+                string displayName = null;
+                string href = null;
+
+                if (block.Inline.First() is LinkInline linkInline)
+                {
+                    if (!string.IsNullOrEmpty(linkInline.Url))
+                        href = linkInline.Url;
+                    if (!string.IsNullOrEmpty(linkInline.Title))
+                        displayName = linkInline.Title;
+
+                    if (linkInline.Any() && linkInline.First() is LiteralInline literalInline)
+                    {
+                        name = literalInline.Content.ToString();
+                    }
+                }
+                else if (block.Inline.First() is LiteralInline literalInline)
+                {
+                    name = literalInline.Content.ToString();
+                }
+
+                var currentItem = new TableOfContentsInputItem
+                {
+                    DisplayName = displayName,
+                    Name = name,
+                    Href = href,
+                };
+
+                return currentItem;
+            }
+        }
+
+        private static MarkdownPipeline CreateMarkdownPipeline()
+        {
+            var markdownContext = new MarkdownContext(null, LogWarning, LogError, null, null);
+
+            return new MarkdownPipelineBuilder()
+                .UseYamlFrontMatter()
+                .UseDocfxExtensions(markdownContext)
+                .UseExtractTocYamlHeader()
+                .Build();
+        }
+
+        private static void LogError(string code, string message, string doc, int line)
+        {
+            t_errors.Add(new Error(ErrorLevel.Error, code, message, doc, new Range(line, 0)));
+        }
+
+        private static void LogWarning(string code, string message, string doc, int line)
+        {
+            t_errors.Add(new Error(ErrorLevel.Warning, code, message, doc, new Range(line, 0)));
+        }
+
+        private static MarkdownPipelineBuilder UseExtractTocYamlHeader(this MarkdownPipelineBuilder builder)
```

Comment:
>UseExtractTocYamlHeader [](start = 47, length = 23)

This method seems to be the same as the one in ExtractYamlHeader.cs? why not just use that one instead?